### PR TITLE
Zone outage message component

### PR DESCRIPTION
**Pull request by corradio**
**Note: This was done as a part of a hackday (our internal monthly hackathons) and the code here is a proof of concept rather than final implementation. Next steps and prioritisation of this is still to be decided!**

## Description

The goal of this PR is to make it faster for us to report data errors and potential outages.
This PR still needs some work (testing, design, styling implementation), so hopefully someone can take over.

The way it works is that it checks a [publicly readable spreadsheet](https://docs.google.com/spreadsheets/d/1C7zULJugzjFj3zw98WZYzBhEIHNYzTtTLFdTcVdzqzc/edit#gid=0) (only editable by the Electricity Maps team) and displays a message accordingly.

cc @mathilde-daugy @unitrium @madsnedergaard @tonypls 

### Preview

image

### CODE REVIEW

File: web/src/features/panels/zone/ZoneHeader.tsx [hunk 1]
```diff
@@ -49,13 +51,56 @@ export function ZoneHeader({
...
+        'message': matching.c[1].v,
+        'issueId': matching.c[2].v,
+      });
+    } else {
+      setOutageData(null);
+    }
+  };
+  useEffect(() => {
+    fetchData();
+  }, [zoneId]);
```
Comment (VIKTORVAV99) [inline web/src/features/panels/zone/ZoneHeader.tsx:76]: Would it not be better to do this in the app-backend and return it as a part of the zone data?

Right now we are introducing an additional dependency on google docs being accessible to every client device and we are making extra network calls. By moving it we can replace thousands of network calls on client device (on a aggregated level) with just one call in the app-backend and there is no requirement for the client device to be able to access google docs.

Comment (corradio) [reply web/src/features/panels/zone/ZoneHeader.tsx:76]: Absolutely! This was just a proof of concept. You are absolutely right.

**Review by madsnedergaard — COMMENTED** [review]
Overall works and looks nice! I have suggested some small design tweaks below :)

**Comment by tonypls** [comment]
> Overall works and looks nice! I have suggested some small design tweaks below - I also have the changes locally and can push them all if you want, but I decided to give you the option of picking between them :) I also added it to the DK-DK2 mock data to make testing locally easier.

Changes look great, want to push em then I'll check them out locally?

**Comment by madsnedergaard** [comment]
> > Overall works and looks nice! I have suggested some small design tweaks below - I also have the changes locally and can push them all if you want, but I decided to give you the option of picking between them :) I also added it to the DK-DK2 mock data to make testing locally easier.
> 
> Changes look great, want to push em then I'll check them out locally?

Done! I also removed the break-all, but feel free to change that if it really needs to be there

**Comment by tonypls** [comment]
> > > Overall works and looks nice! I have suggested some small design tweaks below - I also have the changes locally and can push them all if you want, but I decided to give you the option of picking between them :) I also added it to the DK-DK2 mock data to make testing locally easier.
> > 
> > 
> > Changes look great, want to push em then I'll check them out locally?
> 
> Done! I also removed the break-all, but feel free to change that if it really needs to be there
Here's the case where we want break-all:
image

I'm not sure if it's necessary but it is an edge case I had tested.


I've also updated the overflow, the sizing of the icons and added a div so icons get squished when there is overflow.

image

**Review by madsnedergaard — APPROVED** [review]
Awesome, let's go! 🥳

**Comment by luuu-xu** [comment]
Hi, while I was working on issue 5309, I noticed that our ZoneDetail type does not have the required zoneMessage property. Should we fix that?

**Comment by tonypls** [comment]
> Hi, while I was working on issue 5309, I noticed that our ZoneDetail type does not have the required zoneMessage property. Should we fix that?

That would be great 👍

**Comment by luuu-xu** [comment]
How should we approach this small fix? Opening an issue and PR to fix it?

**Comment by VIKTORVAV99** [comment]
> How should we approach this small fix? Opening an issue and PR to fix it?

Just opening a PR is enough. No need to create a issue if you are fixing it at the same time, at least that's my personal opinion.

**Comment by luuu-xu** [comment]
Oh, I am so sorry. I fetched master branch and realized ZoneDetails does have zoneMessage property. I think when I was working my issue branch I did not diverge from the most recent master commit so TypeScript had an error.

**Comment by VIKTORVAV99** [comment]
> Oh, I am so sorry. I fetched master branch and realized ZoneDetails does have zoneMessage property. I think when I was working my issue branch I did not diverge from the most recent master commit so TypeScript had an error.

Don't worry about it, it happens to us all.

I have done such mistakes myself a more than once occasion. 😅

**Pull request merged**